cpp/dolfinx/mesh/generation.h

```diff
@@ -333,43 +338,42 @@ Mesh<T> build_tet(MPI_Comm comm, MPI_Comm subcomm,
   common::Timer timer("Build BoxMesh (tetrahedra)");
   std::vector<T> x;
   std::vector<std::int64_t> cells;
-  if (subcomm != MPI_COMM_NULL)
-  {
-    x = create_geom<T>(subcomm, p, n);
+  fem::CoordinateElement<T> element(CellType::tetrahedron, 1);
 
-    const std::int64_t nx = n[0];
-    const std::int64_t ny = n[1];
-    const std::int64_t nz = n[2];
-    const std::int64_t n_cells = nx * ny * nz;
+  if (subcomm == MPI_COMM_NULL)
+    return create_mesh(comm, subcomm, cells, element, subcomm, x,
+                       {x.size() / 3, 3}, partitioner);
 
-    std::array range_c = dolfinx::MPI::local_range(
-        dolfinx::MPI::rank(subcomm), n_cells, dolfinx::MPI::size(subcomm));
-    cells.reserve(6 * (range_c[1] - range_c[0]) * 4);
+  x = create_geom<T>(subcomm, p, n);
 
-    // Create tetrahedra
-    for (std::int64_t i = range_c[0]; i < range_c[1]; ++i)
-    {
-      const std::int64_t iz = i / (nx * ny);
-      const std::int64_t j = i % (nx * ny);
-      const std::int64_t iy = j / nx;
-      const std::int64_t ix = j % nx;
-      const std::int64_t v0 = iz * (nx + 1) * (ny + 1) + iy * (nx + 1) + ix;
-      const std::int64_t v1 = v0 + 1;
-      const std::int64_t v2 = v0 + (nx + 1);
-      const std::int64_t v3 = v1 + (nx + 1);
-      const std::int64_t v4 = v0 + (nx + 1) * (ny + 1);
-      const std::int64_t v5 = v1 + (nx + 1) * (ny + 1);
-      const std::int64_t v6 = v2 + (nx + 1) * (ny + 1);
-      const std::int64_t v7 = v3 + (nx + 1) * (ny + 1);
-
-      // Note that v0 < v1 < v2 < v3 < vmid
-      cells.insert(cells.end(),
-                   {v0, v1, v3, v7, v0, v1, v7, v5, v0, v5, v7, v4,
-                    v0, v3, v2, v7, v0, v6, v4, v7, v0, v2, v6, v7});
-    }
+  const auto [nx, ny, nz] = n;
+  const std::int64_t n_cells = nx * ny * nz;
+
+  std::array range_c = dolfinx::MPI::local_range(
+      dolfinx::MPI::rank(subcomm), n_cells, dolfinx::MPI::size(subcomm));
+  cells.reserve(6 * (range_c[1] - range_c[0]) * 4);
+
+  // Create tetrahedra
+  for (std::int64_t i = range_c[0]; i < range_c[1]; ++i)
+  {
+    const std::int64_t iz = i / (nx * ny);
```

Comment:
We usually dont use const assignment for integers. Maybe @garth-wells can comment?

Reply:
House style is we don't normally use `const` for owned variables inside functions.

Reply:
So remove `const` in these cases? They were marked `const` before this PR.